include/swift/AST/SourceFile.h

```diff
@@ -662,6 +662,14 @@ class SourceFile final : public FileUnit {
     Scope = nullptr;
   }
 
+  /// Whether the given string matches the file ID of this file.
+  bool matchesFileID(StringRef fileID) const;
+
+  /// Parses a file ID string into the module name and file name (with ".swift"
+  /// suffix). Returns nullopt if the file ID string is invalid.
+  static std::optional<std::pair<StringRef, StringRef>>
```

Comment:
I think if we had a `struct FileIdStr { StringRef module, file; };` that would make most of this comment redundant.  Up to you, I have no strong feelings here.

Reply:
This might give the opportunity to implement `matchesFileID` as an `operator==` on `FileIdStr`.

Reply:
I created a `FileIDStr` struct, but I decided not to use `operator==` since `matches` is more explicit (and still fairly succinct)